sstables/sstable_directory.hh

```diff
@@ -49,7 +49,12 @@ public:
 
     template <std::ranges::range Container, typename Func>
     requires std::is_invocable_r_v<future<>, Func, typename std::ranges::range_value_t<Container>&>
-    future<> parallel_for_each(Container& C, Func func);
```

Comment:
Could we use `std::function` instead? How sensitive are the callers to an extra vptr dereference?

Reply:
On the other hand, it still need Container template parameter thus still needs to be a template, so what's the point in std::function?

Reply:
Ok, fair enough.